### PR TITLE
Compile time option to run L0 Sampling code and better GraphZeppelin configuration

### PR DESCRIPTION
**Pull request by etwest**
The current code runs a support finding algorithm that is sufficient for CC but maybe not for other relevant graph algorithms. This pull request adds a compile time option to run the code that is provably an L0 sampling algorithm.

Additionally, it reflects changes made to the Guttering System configuration to allow for easy specification of the batch size as a function of the Supernode size. This is especially relevant when used in combination with a modified number of sketches.

### CODE REVIEW

**Comment by etwest** [comment]
Note, merge pull request in GutterTree repo first and switch branch tag back to main before merging.

File: include/supernode.h [hunk 1]
```diff
@@ -210,7 +210,13 @@ class Supernode {
...
 
-  static constexpr size_t default_fail_factor = 4;
+#ifdef L0_SAMPLING
+  static constexpr size_t default_fail_factor = 128;
+  static constexpr double num_sketches_div = log2(3) - 1;
+#else
+  static constexpr size_t default_fail_factor = 2;
+  static constexpr double num_sketches_div = log2(4) - log2(3);
+#endif
 };
```
Comment (DanielDeLayo) [inline include/supernode.h:221]: direct number of cols instead

File: src/l0_sampling/sketch.cpp [hunk 2]
```diff
@@ -72,6 +72,25 @@ Sketch::Sketch(const Sketch& s) : seed(s.seed) {
...
+    col_hash_t depth = Bucket_Boruvka::get_index_depth(update_idx, column_seed(i), num_guesses);
+    likely_if(depth < num_guesses) {
+      for (col_hash_t j = 0; j <= depth; ++j) {
+        size_t bucket_id = i * num_guesses + j;
+        Bucket_Boruvka::update(bucket_a[bucket_id], bucket_c[bucket_id], update_idx, checksum);
+      }
+    }
+  }
+}
+#else // Use support finding algorithm instead. Faster but no guarantee of uniform sample.
```
Comment (DanielDeLayo) [inline src/l0_sampling/sketch.cpp:92]: support finding algorithm?

Comment (DanielDeLayo) [inline src/l0_sampling/sketch.cpp:92]: support finding algorithm?

**Review by DanielDeLayo — CHANGES_REQUESTED** [review]

**Review by DanielDeLayo — APPROVED** [review]

**Pull request merged**